### PR TITLE
AMC_BLDC: Alignment with the latest changes on MBD

### PR DESCRIPTION
**Pull request by sgiraz**
What changes in this PR:
- The mbd I/O interfaces have been updated with the latest architectural model changes.
- Workarounds for `STATUS` CAN packet removed because it is now managed internally to the `can_encoder`
- MBDAgent updated with minor improvements and now it is predisposed to manage multi I/O CAN packets although it still elaborates one packet at a time instance in RX

**Notes:**
- ⚠️ The EMS still sends 1 target setpoint every 2ms instead of every 1ms. Remember that this workaround allows the AMC_BLDC to consume all its incoming packages. This is a temporary workaround until the multipackage manager will be completed inside the `mbd`.
- ⚠️ After the fault button is pressed and then released, if we send the `SET_CONTROL_MODE` IDLE through the `yarpmotorgui`, there are 2 consecutive `SET_CONTROL_MODE` message packets are sent
- ✅ This PR has been successfully tested on `wrist-setup`.

### CODE REVIEW

File: emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp [hunk 1]
```diff
@@ -390,6 +398,10 @@ bool embot::app::application::theMBDagent::Impl::tick(std::vector<embot::prot::c
         }
     }
     
+    // If motor configuration parameters changed due to a SET_MOTOR_CONFIG message, then update hal as well (Only pole_pairs at the moment)
+    // TODO: When should perform the following update within the mbd after a SET_MOTOR_CONFIG message has been received
+    MainConf.pwm.poles = InitConfParams.motorconfig.pole_pairs;
```
Comment (pattacini) [inline emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp:403]: Hi @sgiraz 

This is not correct.
It shall be instead something like:
```c++
MainConf.pwm.poles = ConfigurationParameters.motorconfig.pole_pairs;
```

File: emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp [hunk 2]
```diff
@@ -431,7 +443,7 @@ void embot::app::application::theMBDagent::Impl::onCurrents_FOC_innerloop(void *
     electricalAngleOld = electricalAngle;
     
     // calculate the current joint position
-    position = position + delta / 7;  // the motor has 14 poles, hence 7 pole pairs
+    position = position + delta / InitConfParams.motorconfig.pole_pairs;  // the motor has 14 poles, hence 7 pole pairs
```
Comment (pattacini) [inline emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp:446]: Same here as above:

It shall be instead something like:
```c++
position = position + delta / ConfigurationParameters.motorconfig.pole_pairs;
```

**Review by pattacini — CHANGES_REQUESTED** [review]
Hi @sgiraz 

There're a couple of points to be addressed.

File: emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp (hunk 1 above)
Comment (pattacini) [reply emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp:403]: `ConfigurationParameters` is the model's output.

Comment (sgiraz) [reply emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp]: Nice catch @pattacini! 
Resolved!

File: emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp (hunk 2 above)
Comment (sgiraz) [reply emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp]: Okay, done!

**Review by pattacini — APPROVED** [review]

File: emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp (hunk 2 above)
Comment (pattacini) [reply emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp]: Thanks!

File: emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp (hunk 1 above)
Comment (pattacini) [reply emBODY/eBcode/arch-arm/board/amcbldc/application06/src/embot_app_application_theMBDagent.cpp]: Thanks!

**Review by marcoaccame — APPROVED** [review]
ok

**Pull request merged**